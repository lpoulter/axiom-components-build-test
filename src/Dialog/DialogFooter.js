import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';

export default class DialogFooter extends Component {
  static propTypes = {
    /** Footer content inside the Dialog, a good place for some buttons */
    children: PropTypes.node,
    /** Class name to be appended to the element */
    className: PropTypes.string,
  };

  render() {
    const { children, className, ...rest } = this.props;

    const classes = classnames('ax-dialog__footer', className);

    return (
      <Base { ...rest } className={ classes }>
        { children }
      </Base>
    );
  }
}
