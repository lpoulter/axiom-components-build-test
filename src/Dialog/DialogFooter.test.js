import React from 'react';
import renderer from 'react-test-renderer';
import DialogFooter from './DialogFooter';

const getComponent = (props = {}) =>
  renderer.create(
    <DialogFooter { ...props }>
      Lorem ipsum
    </DialogFooter>
  );

describe('DialogFooter', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
