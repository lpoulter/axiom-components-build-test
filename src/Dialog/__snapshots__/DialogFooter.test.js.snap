// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogFooter renders with defaultProps 1`] = `
<div
  className="ax-dialog__footer"
>
  Lorem ipsum
</div>
`;
